packages/beacon-node/src/chain/opPools/aggregatedAttestationPool.ts

```diff
@@ -359,40 +393,83 @@ export class AggregatedAttestationPool {
+                totalNewSeenEffectiveBalance: 0,
+                newSeenAttesters: 0,
+                notSeenAttesters: 0,
+                committeeMembers: 0,
               };
             }
-            sameAttDataCons[i].byCommittee.set(committeeIndex, attestationNonParticipation);
-            sameAttDataCons[i].totalNotSeenCount += attestationNonParticipation.notSeenAttesterCount;
+            const sameAttDataCon = sameAttDataCons[i];
+            // committeeIndex was from a map so it should be unique, but just in case
+            if (!sameAttDataCon.byCommittee.has(committeeIndex)) {
+              sameAttDataCon.byCommittee.set(committeeIndex, attestationNonParticipation);
+              sameAttDataCon.totalNewSeenEffectiveBalance += attestationNonParticipation.newSeenEffectiveBalance;
+              sameAttDataCon.newSeenAttesters += attestationNonParticipation.newSeenAttesters;
+              sameAttDataCon.notSeenAttesters += attestationNonParticipation.notSeenAttendingIndices.size;
+              sameAttDataCon.committeeMembers += attestationGroup.committee.length;
+            }
           }
         } // all committees are processed
 
         // after all committees are processed, we have a list of sameAttDataCons
         for (const consolidation of sameAttDataCons) {
-          const score = consolidation.totalNotSeenCount / slotDelta;
+          const score = consolidation.totalNewSeenEffectiveBalance / slotDelta;
           consolidations.set(consolidation, score);
-
           // Stop accumulating attestations there are enough that may have good scoring
           if (consolidations.size >= MAX_ATTESTATIONS_ELECTRA * 2) {
+            stopReason = ScannedSlotsTerminationReason.MaxConsolidationReached;
             break slot;
           }
         }
       }
+
+      // finished processing a slot
+      scannedSlots++;
     }
+
+    this.metrics?.opPool.aggregatedAttestationPool.packedAttestations.totalConsolidations.set(consolidations.size);
+
     const sortedConsolidationsByScore = Array.from(consolidations.entries())
       .sort((a, b) => b[1] - a[1])
       .map(([consolidation, _]) => consolidation)
       .slice(0, MAX_ATTESTATIONS_ELECTRA);
 
     // on chain aggregation is expensive, only do it after all
-    return sortedConsolidationsByScore.map(aggregateConsolidation);
+    const packedAttestationsMetrics = this.metrics?.opPool.aggregatedAttestationPool.packedAttestations;
+    const packedAttestations: electra.Attestation[] = new Array(sortedConsolidationsByScore.length);
+    for (const [i, consolidation] of sortedConsolidationsByScore.entries()) {
+      packedAttestations[i] = aggregateConsolidation(consolidation);
+
+      // record metrics of packed attestations
+      const committeeCount = consolidation.byCommittee.size;
+      packedAttestationsMetrics?.committeeBits.set({index: i}, committeeCount);
+      packedAttestationsMetrics?.committeeMembers.set({index: i}, consolidation.committeeMembers);
```

Comment:
so the difference between these to is actual (`committeeBits`) included vs. theoretical maximum (`committeeMembers`)?

Reply:
> so the difference between these to is actual (`committeeBits`) included vs. theoretical maximum (`committeeMembers`)?

`committeeBits` is the number of committee. `committeeMembers` is total number of validators across all included committees (think of number of bits in `aggregationBits` in the final aggregated attestation).

Reply:
if `committeeMembers` is all validators across all committees than the name of the metric seems misleading to me

Reply:
will change to `crossCommiteeMembers` and add comment for it

Reply:
> will change to `crossCommiteeMembers` and add comment for it

why not call it `totalValidatorCount` or `totalValidators`, I guess I am mostly confused why we talk about committee members here even though it's consolidated across committees

Reply:
that sounds good to me too 👍

Reply:
called it `totalAttesters`, we seem to use the term validator and attester interchangeably here in a few places, however attester seems to be more common, this part is not that important but at least what confused me a bit is if we don't use the correct terms when we talk about just committee members vs. attesters across all committees